ceno_zkvm/src/structs.rs

```diff
@@ -244,6 +284,7 @@ impl<E: ExtensionField> ZKVMFixedTraces<E> {
 
 #[derive(Default, Clone)]
 pub struct ZKVMWitnesses<E: ExtensionField> {
+    keccak_trace: <KeccakLayout<E> as gkr_iop::ProtocolWitnessGenerator<E>>::Trace,
```

Comment:
It seems like other members are `E::BaseField` elements. Does it make sense that we also make it `Vec<Vec<E::BaseField>>`, which is the result after calling `phase1_witness()`?